### PR TITLE
Banking stage refactor commit transactions

### PR DESCRIPTION
**Pull request by sakridge**
#### Problem

Really long functions in banking-stage are hard to understand.

#### Summary of Changes

Break them up a bit and combine like-data in structs. No functional change.

Fixes #

### CODE REVIEW

File: core/src/banking_stage.rs [hunk 1]
```diff
@@ -1202,6 +1213,129 @@ impl BankingStage {
         }
     }
 
+    #[allow(clippy::too_many_arguments)]
+    fn commit_transactions(
```
Comment (apfitzge) [inline core/src/banking_stage.rs:1217]: Love to see someone break these monster functions up. Might take it one step further, this function is still fairly large and seems to be doing 3 three distinct things:
1. commiting transactions
2. collecting commit statuses from commit results (1)
3. collect balances and send from commit results (1)

Can either rework this PR or do a follow up PR if you agree - could potentially do something like:
```
fn commit_transactions(...) -> Vec<CommitTransactionDetails>;
fn get_commit_status(tx_results) -> Vec<CommitTransactionDetails>;
fn collect_balances_and_send_status_batch(...);

fn commit_and_send_status_batch() -> (commit_time_us, Vec<CommitTransactionDetails>); // Combination of the above 3 functions to replace this one
```

**Review by apfitzge — APPROVED** [review]
looks good - no behavior change, code is cleaner. Left one comment about potentially breaking it up further.

**Review by taozhu-chicago — APPROVED** [review]
lgtm, thanks for cleaning it

File: core/src/banking_stage.rs (hunk 1 above)
Comment (sakridge) [reply core/src/banking_stage.rs:1217]: yea, it's a good idea. I think a follow-up would be the best way.

**Pull request merged**